gateway-developer-plugin/src/main/java/com/ca/apim/gateway/cagatewayconfig/BuildDeploymentBundleTask.java

```diff
@@ -61,7 +70,32 @@ public void perform() {
         BundleFileBuilder bundleFileBuilder = InjectionRegistry.getInjector().getInstance(BundleFileBuilder.class);
         final ProjectInfo projectInfo = new ProjectInfo(getProject().getName(), getProject().getGroup().toString(),
                 getProject().getVersion().toString());
+        final List<DependentBundle> dependentBundles = getDependentBundles(dependencies.getFiles());
         bundleFileBuilder.buildBundle(from.isPresent() ? from.getAsFile().get() : null, into.getAsFile().get(),
-                new ArrayList<>(dependencies.getFiles()), projectInfo);
+                dependentBundles, projectInfo);
+    }
+
+    private List<DependentBundle> getDependentBundles(Set<File> files) {
+        final List<DependentBundle> dependentBundles = new ArrayList<>();
+        ConfigurationContainer configurations = getProject().getConfigurations();
+        Configuration configuration = configurations.getByName("bundle");
+        DependencySet dependencySet = configuration.getDependencies();
+
+        files.forEach(file -> {
+            if (file.getName().endsWith(BUNDLE_EXTENSION) || file.getName().endsWith(JsonFileUtils.METADATA_FILE_NAME_SUFFIX)) {
```

Comment:
Shouldn't this be `INSTALL_BUNDLE_EXTENSION` instead of `BUNDLE_EXTENSION`? Do we need delete bundle as well?

Reply:
we dont get delete bundle as artifact, its the dependency definied in the build.gradle